### DISCUSSION TITLE
[FR]: Use LSP to support more languages

### DISCUSSION THREAD

**Discussion by jsharpe** [aspect-build/rules_lint#312] [Ideas]
### Describe the feature

Mason.nvim is a plugin manager for NeoVim that contains a library of LSP compatible formatters and linters.
It draws from the registry here: https://github.com/mason-org/mason-registry/tree/main

Being able to pull this data and automatically add support for language formatters / linters from this library of formatters and linters would massively expand the number of languages that are supported by rules_lint.

I've not looked into the feasibility of this but I suspect that we could modify rules_lint to provide a wrapper that supports the LSP protocol that is used by VSCode / NeoVim etc.. and provide a generic linting / formatting capability via this.

**Comment by alexeagle** [comment]
Another advantage of LSP aside from expanding languages may be performance.

For example pyright `--watch` mode may be needed for performance: https://github.com/microsoft/pyright/discussions/4809
We also have a client who had to fork the eslint aspect here in order to allow it to use its local caching.

Of course this is a tradeoff with hermeticity, the same as persistent workers for compilation under Bazel, very easy for them to get stateful and have bugs like passing when they should fail after removing an input.
